skimage/transform/tests/test_geometric.py

```diff
@@ -1077,3 +1212,69 @@ def test_2D_only_implementations():
         _ = tf.rotation
     with pytest.raises(NotImplementedError):
         _ = tf.shear
+
+
+@pytest.mark.parametrize('tform_class', HMAT_TFORMS)
+def test_kw_only_params(tform_class):
+    # Check only matrix can be passed as positional arg.
+    with pytest.raises(TypeError):
+        tform_class(None, None)
+
+
+def test_kw_only_emt():
+    # Check all parameters are keyword only for EssentialMatrixTransform.
+    with pytest.raises(TypeError):
+        EssentialMatrixTransform(None)
+
+
+@pytest.mark.parametrize('tform_class', HMAT_TFORMS)
+def test_init_contract_dims(tform_class):
+    allows_nd = tform_class in HMAT_TFORMS_ND
+    # 2D identity is default.
+    for tf in (
+        tform_class(),
+        tform_class(dimensionality=2),
+        tform_class.identity(),
+        tform_class.identity(None),
+        tform_class(None),
+    ):
+        assert_equal(tf.params, np.eye(3))
+    ok_dims = (2, 3, 4, 5) if allows_nd else (2,)
+    for d in ok_dims:
+        h_d = d + 1
+        # Identity for acceptable dimensions.
+        for tf in (
+            tform_class(dimensionality=d),
+            tform_class.identity(d),
+            tform_class.identity(dimensionality=d),
+        ):
+            assert_equal(tf.params, np.eye(h_d))
+        # Wrong shape for given dimensions.
+        for matrix in np.eye(h_d)[:-1], np.eye(h_d)[:, :-1]:
+            with pytest.raises(ValueError):
+                tform_class(matrix)
+    err_dims = (1,) if allows_nd else (1, 3, 4, 5)
+    for matrix in [np.eye(d + 1) for d in err_dims]:
+        with pytest.raises(NotImplementedError):
+            tform_class(matrix)
+    # Test vector matrix input invalid.
+    with pytest.raises(ValueError):
+        tform_class(np.zeros((2, 3)))
+
+
+def test_broadcasting():
+    # Scalar scale broadcasts.
+    translation = [3, 4, 5]
+    tf = SimilarityTransform(scale=2, translation=translation)
+    assert_equal(tf.params, _from_matvec(np.eye(3) * 2, translation))
+    # Translation does broadcast.
+    # 2D.
+    tf = SimilarityTransform(scale=2, translation=10)
+    assert_equal(tf.params, _from_matvec(np.eye(2) * 2, [10, 10]))
+    # 3D.
+    tf = SimilarityTransform(scale=[2, 3, 4], translation=10)
+    assert_equal(tf.params, _from_matvec(np.diag([2, 3, 4]), [10] * 3))
+    # Scalar rotation does not broadclast
```

Comment:
```suggestion
    # Scalar rotation does not broadcast.
```
I wouldn't usually correct this, but just to be consistent with the above...